docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+For this example, we want to create a secure box called `private_timer`. The `private_timer` box has exclusive access to the PIT timer and to the GPIO PORT C on the NXP FRDM-K64F board. These peripherals cannot be accessed from outside the secure box.
+
+Each secure box must have at least one thread, which we call the box main thread. In our `private_timer` box we will only use this thread throughout the whole program. The thread will constantly save the current timer value in a private buffer. In addition, we want to print the content of the buffered timer values whenever we press the button on the board.
+
+We want the box to have exclusive access to the following resources:
+
+* The timer and push-button peripherals (peripheral ACLs). Nobody else should be able to read the timer values.
+* The push-button interrupt (IRQ ACL). We want the button IRQ to be re-routed to our box-specific ISR.
+* The buffer that holds the timer samples (dynamic memory ACL).
+* The static memory that holds information about the timer buffer (static memory ACL).
+
+Create a new source file, `~/code/uvisor-example/source/secure_box.cpp`. We will configure the secure box inside this file. The secure box name for this example is `private_timer`.
+
+```C
+/* ~/code/uvisor-example/source/secure_box.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+#include "uvisor-lib/uvisor-lib.h"
+
+/* Private static memory for the secure box */
+typedef struct {
+    uint32_t * buffer;
+    int index;
+} PrivateTimerStaticMemory;
+
+/* ACLs list for the secure box: Timer (PIT). */
+static const UvisorBoxAclItem g_private_timer_acls[] = {
+    {PIT,   sizeof(*PIT),   UVISOR_TACLDEF_PERIPH}
+    {PORTC, sizeof(*PORTC), UVISOR_TACLDEF_PERIPH},
+};
+
+static void private_timer_main_thread(const void *);
+
+/* Secure box configuration */
+UVISOR_BOX_NAMESPACE(NULL);                   /* Box namespace. It can be NULL for this example. */
+UVISOR_BOX_HEAPSIZE(4096);                    /* Heap size for the secure box */
+UVISOR_BOX_MAIN(private_timer_main_thread,    /* Secure box main thread function */
+                osPriorityNormal,             /* Priority of the secure box main thread */
+                1024);                        /* Stack size for the secure box main thread */
+UVISOR_BOX_CONFIG(private_timer,              /* Name of the secure box */
+                  g_private_timer_acls,       /* ACLs list for the secure box */
+                  1024,                       /* Stack size for the secure box */
+                  PrivateTimerStaticMemory);  /* Private static memory for the secure box. */
+```
+
+### Create the secure box main thread function
+
+In general, you can decide what to do in your box main thread. You can run it once and then kill it, use it to configure memories, peripherals, or to create other threads. In this app, the box main thread is the only thread for the `private_timer` box, and it will run throughout the whole program.
```

Comment:
Didn't test, but it *shouldn't be a problem*™. The box's main thread is never used in any other way.